atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/backup/CassandraRepairHelper.java

```diff
@@ -0,0 +1,120 @@
+        KeyValueService kvs = keyValueServiceFactory.apply(namespace);
+        CqlCluster cqlCluster = getCqlCluster(namespace);
+        kvs.getAllTableNames().stream()
+                .filter(TABLES_TO_REPAIR::contains)
+                .map(TableReference::getTableName)
+                .map(tableName -> getRangesToRepair(cqlCluster, namespace, tableName))
+                // TODO(gs): this will do repairs serially, instead of batched. Is this fine? Port batching from
+                //   internal product?
+                .forEach(repairTable);
+    }
+
+    private CqlCluster getCqlCluster(Namespace namespace) {
+        CassandraKeyValueServiceConfig config = keyValueServiceConfigFactory.apply(namespace);
+        return CqlCluster.create(config);
+    }
+
+    // VisibleForTesting
+    public Map<InetSocketAddress, Set<LightweightOppTokenRange>> getRangesToRepair(
+            CqlCluster cqlCluster, Namespace namespace, String tableName) {
+        Map<InetSocketAddress, Set<TokenRange>> tokenRanges = getTokenRangesToRepair(cqlCluster, namespace, tableName);
+        return KeyedStream.stream(tokenRanges).map(this::makeLightweight).collectToMap();
+    }
+
+    private Map<InetSocketAddress, Set<TokenRange>> getTokenRangesToRepair(
+            CqlCluster cqlCluster, Namespace namespace, String tableName) {
+        String cassandraTableName = getCassandraTableName(namespace, tableName);
+        return cqlCluster.getTokenRanges(cassandraTableName);
+    }
+
+    private String getCassandraTableName(Namespace namespace, String tableName) {
+        TableReference tableRef = TableReference.create(toKvNamespace(namespace), tableName);
+        return AbstractKeyValueService.internalTableName(tableRef);
+    }
+
+    private com.palantir.atlasdb.keyvalue.api.Namespace toKvNamespace(Namespace namespace) {
+        return com.palantir.atlasdb.keyvalue.api.Namespace.create(namespace.get());
+    }
+
+    private Set<LightweightOppTokenRange> makeLightweight(Set<TokenRange> tokenRanges) {
+        return tokenRanges.stream().flatMap(this::makeLightweight).collect(Collectors.toSet());
+    }
+
+    private Stream<LightweightOppTokenRange> makeLightweight(TokenRange tokenRange) {
+        LightweightOppToken startToken = LightweightOppToken.serialise(tokenRange.getStart());
+        LightweightOppToken endToken = LightweightOppToken.serialise(tokenRange.getEnd());
+
+        if (startToken.compareTo(endToken) <= 0) {
+            return Stream.of(LightweightOppTokenRange.of(startToken, endToken));
+        } else {
+            // Handle wrap-around
+            LightweightOppToken unbounded = unboundedToken();
+            LightweightOppTokenRange greaterThan = LightweightOppTokenRange.of(startToken, unbounded);
+            LightweightOppTokenRange atMost = LightweightOppTokenRange.of(unbounded, endToken);
+            return Stream.of(greaterThan, atMost);
+        }
+    }
+
+    private LightweightOppToken unboundedToken() {
```

Comment:
This is a bit magical, where did we come up with it from?

Reply:
cassandra-driver-core: https://github.com/datastax/java-driver/blob/3.x/driver-core/src/main/java/com/datastax/driver/core/Token.java#L352